Cargo.toml

```diff
@@ -39,11 +39,12 @@ decimal128 = ["decimal"]
 name = "bson"
 
 [dependencies]
+ahash = "0.7.2"
```

Comment:
The `ahash` crate provides a fast yet safe hashing algorithm. In fact, I think it's now the default one used by `std::HashMap`: https://github.com/rust-lang/hashbrown/blob/master/Cargo.toml. In 2.0, `indexmap` looks like they will be updating their default to use it, so we can remove this dependency then (https://github.com/bluss/indexmap/issues/135)